From aa2172de1af414bb296f8aa5446f5d5adc2123bf Mon Sep 17 00:00:00 2001
From: Chiara Marmo <marmochiaskl@gmail.com>
Date: Tue, 20 Sep 2022 09:59:16 -1000
Subject: [PATCH 1/4] Fix ranking for scipy >= 1.10.

---
 sklearn/model_selection/_search.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/sklearn/model_selection/_search.py b/sklearn/model_selection/_search.py
index 7cf2307ac85e0..5bb892f57c66f 100644
--- a/sklearn/model_selection/_search.py
+++ b/sklearn/model_selection/_search.py
@@ -965,11 +965,13 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
             results["std_%s" % key_name] = array_stds
 
             if rank:
-                rank_result = rankdata(-array_means, method="min")
                 # when input is nan, scipy >= 1.10 rankdata returns nan. To
                 # keep previous behaviour nans are set to the
-                # maximum possible rank
-                rank_result[np.isnan(rank_result)] = len(rank_result)
+                # minimum value in the array before ranking
+                array_means = np.asarray(
+                    [min(array_means) - 1 if np.isnan(a) else a for a in array_means]
+                )
+                rank_result = rankdata(-array_means, method="min")
                 rank_result = np.asarray(rank_result, dtype=np.int32)
                 results["rank_%s" % key_name] = rank_result
