--- sklearn/model_selection/_search.orig.py	2023-05-05 19:59:15
+++ sklearn/model_selection/_search.py	2023-05-05 19:59:31
@@ -965,9 +965,14 @@
             results["std_%s" % key_name] = array_stds
 
             if rank:
-                results["rank_%s" % key_name] = np.asarray(
-                    rankdata(-array_means, method="min"), dtype=np.int32
-                )
+                # when input is nan, scipy >= 1.10 rankdata returns nan. To
+                # keep previous behaviour nans are set to be smaller than the
+                # minimum value in the array before ranking
+                min_array_means = min(array_means) - 1
+                array_means = np.nan_to_num(array_means, copy=True, nan=min_array_means)
+                rank_result = rankdata(-array_means, method="min")
+                rank_result = np.asarray(rank_result, dtype=np.int32)
+                results["rank_%s" % key_name] = rank_result
 
         _store("fit_time", out["fit_time"])
         _store("score_time", out["score_time"])
